### ISSUE TITLE
Wrong socket permission since 1.6.27

### ISSUE THREAD

**Issue by weirdan** [GoogleCloudPlatform/google-fluentd#233]
Ever since Jan 14 google-fluentd fails to set proper permission on unix socket created by `@type unix` source:
```console
$ ls -l /var/run/google-fluentd/google-fluentd.sock
srwxr-xr-x 1 root root 0 Jan 14 04:28 /var/run/google-fluentd/google-fluentd.sock
```
```console
$ cat /etc/google-fluentd/config.d/socket.conf
<source>
  @type unix
  path /var/run/google-fluentd/google-fluentd.sock
</source>
```
This looks similar to fluent/fluentd#1019

**Comment by GLStephen** [comment]
We are also seeing this issue. This is a pretty fundamental issue.

**Comment by GLStephen** [comment]
@jkohen I can confirm that manually rolling back this change resolves the issue with permissions on the socket file.

https://github.com/GoogleCloudPlatform/google-fluentd/pull/225

**Comment by qingling128** [comment]
Took a quick look.

It seems like that for standalone worker mode, Fluentd is setting `chuser` and `chgroup`, but not `chumask`.

https://github.com/fluent/fluentd/blob/c9dde7104ea8dfc2442fd73615bddac2d8c5f6c1/lib/fluent/supervisor.rb#L596

Filed a bug with the upstream Fluentd repo: https://github.com/fluent/fluentd/issues/2984.

For the workaround, we can run the agent in supervisor mode.

**Comment by qingling128** [comment]
Looks like the upstream issue has been fixed by https://github.com/fluent/fluentd/pull/2987.

Next step: Wait for a Fluentd release and upgrade to that version.

**Comment by nicolaiskogheim** [comment]
The fix was [released in v1.10.4](https://github.com/fluent/fluentd/commit/5a2afdccc7fc3766f0728c29b04519784ff7ef3f) btw.

**Comment by jkschulz** [comment]
This is fixed by the upgrade to fluentd v1.11.2 in #283, which will be released in `google-fluentd` 1.8.0:

```
$ ls -l /var/run/google-fluentd/google-fluentd.sock
srwxrwxrwx 1 root root 0 Aug 12 16:36 /var/run/google-fluentd/google-fluentd.sock
```

**Comment by jkschulz** [comment]
This fix was released in [1.8.0 on Aug 17](https://github.com/GoogleCloudPlatform/google-fluentd/releases/tag/1.8.0).
